Moved the generation of the eaw table in libmbfl to CMake time

Orvid opened the pull request:
Moved the generation of the eaw table in libmbfl to CMake time rather than build time, due to complications caused by doing it at build time on Windows.

jwatzman commented:
Does this set the correct inputs and outputs, such that if the input file changes, we regenerate the output, but that we do not do it on every single `make` invocation?

Orvid commented:
As this is done during the CMake configure process, it's never run during a `make` invocation at all with this PR. Previously it was run on the `make` calls.

jwatzman commented:
@JoelMarcey wasn't there some reason we did it during `make` instead of during configuration, I remember there being a diff or debate or something.

@Orvid does CMake get re-run if we update the file (i.e., when we update the submodule), or is this file generated once and that's it?

jwatzman commented:
1d387ce2a33fc75d25c7ee3318537296387d8799 is when we changed it -- apparently was causing problems with `make clean` (which makes sense, since it was done as part of the build process).

I'm not a huge fan of this, doing it during `make` seems like the right thing to me. What are the problems on Windows?

Orvid commented:
This was part of my very early efforts of porting to windows, and, with what I've learned since then, I now know why I changed it. The original CMake file was explicitly running the script with sh, which Windows doesn't have. CMake however will use my mysysgit bash shell to run shell scripts if I've launched cmake from it, which I do as part of a short cleanup and reconfigure script (it copies in a manually created cmake cache which sets all the variables I need as well as speeds up the configure process a lot by setting the variables for whether certain headers and functions are available). CMake will only do this however if you are invoking the shell script directly in the command. As this was originally calling `sh` directly, CMake wouldn't do this, which made me simply switch to calling it directly.

Based on your comments, and what I've learned since this change was originally done, the correct way to fix this on windows is to simply remove the `sh` in the command, and call the bash script directly.

I'll close this PR for now, and re-open it once I've updated it to do this correctly.

Orvid commented:
Alright, so it looks like I had a bit more of an idea what I was doing than I thought. On windows, `gawk` is simply segfaulting when it's called during the build. My first guess is that a dll somewhere is conflicting. I'll dig a bit more and re-open this if I can't figure out any other way to get it to work.
